tests/beaker_tests/cisco_itd_device_group/test_itd_device_group.rb

```diff
@@ -0,0 +1,161 @@
+###############################################################################
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+#
+# See README-develop-beaker-scripts.md (Section: Test Script Variable Reference)
+# for information regarding:
+#  - test script general prequisites
+#  - command return codes
+#  - A description of the 'tests' hash and its usage
+#
+###############################################################################
+
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# Test hash top-level keys
+tests = {
+  master:           master,
+  agent:            agent,
+  operating_system: 'nexus',
+  resource_name:    'cisco_itd_device_group',
+}
+
+# Test hash test cases
+tests[:default] = {
+  desc:           '1.1 Defaults',
+  title_pattern:  'icmpGroup',
+  preclean:       'cisco_itd_device_group',
+  manifest_props: {
+    probe_type: 'default'
+  },
+  code:           [0, 2],
+  resource:       {
+    'probe_type' => 'false'
+  },
+}
+
+tests[:default_icmp] = {
+  desc:           '1.2 Defaults for type icmp',
+  title_pattern:  'icmpGroup',
+  preclean:       'cisco_itd_device_group',
+  manifest_props: {
+    probe_frequency:  'default',
+    probe_retry_down: 'default',
+    probe_retry_up:   'default',
+    probe_timeout:    'default',
+    probe_type:       'icmp',
+  },
+  code:           [0, 2],
+  resource:       {
+    'probe_frequency'  => '10',
+    'probe_retry_down' => '3',
+    'probe_retry_up'   => '3',
+    'probe_timeout'    => '5',
```

Comment:
No, whatever we do not provide in the resource ( that is if the manifest and resource are same), will be taken from manifest directly. So it is checking for icmp as probe type since it is not given in the resource.